src/Agent.Worker/ExecutionContext.cs

```diff
@@ -496,10 +496,16 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
                 }
             }
 
+            message.Variables.TryGetValue("AZP_EXPAND_VULNERABLE_VARIABLES", out var expandVulnerableVariablesKnob);
+
+            string expandVulnerableVariablesKnobValue = expandVulnerableVariablesKnob?.Value ?? Environment.GetEnvironmentVariable("AZP_EXPAND_VULNERABLE_VARIABLES");
+            bool.TryParse(expandVulnerableVariablesKnobValue, out var canExpandVulnerableVariables);
```

Comment:
>TryParse [](http://example.com/codeflow?start=17&length=8)

[suggestion/not blocking]

Better to trace the value !TryParse as I believe that would not be expected

Reply:
I updated the logic, please check